src/bandmap.c

```diff
@@ -656,6 +656,54 @@ freq_t bm_get_center(int band, int mode) {
     return centerfrequency;
 }
 
+/*
+ * filter 'allspots' list according to settings and prepare 'spots' array with
+ * selected spots
+ */
+void filter_spots() {
+    GList *list;
+    spot *data;
+    bool dupe, multi;
+    /* acquire mutex
+     * do not add new spots to allspots during
+     * - aging and
+     * - filtering
+     * furthermore do not allow call lookup as long as
+     * filtered spot array is build anew */
+
+    pthread_mutex_lock(&bm_mutex);
+
+    if (spots)
+	g_ptr_array_free(spots, TRUE);		/* free spot array */
+						/* allocate new one */
+    spots = g_ptr_array_new_full(128, (GDestroyNotify)free_spot);
+
+    list = allspots;
+
+    while (list) {
+	data = list->data;
+
+	/* if spot is allband or allmode is set or band or mode matches
+	 * actual one than add it to the filtered 'spot' array
+	 * drop spots on WARC bands if in contest mode
+	 */
+	multi = bm_ismulti(NULL, data, data->band);
+	dupe = bm_isdupe(data->call, data->band);
+
+	if ((!iscontest || !IsWarcIndex(data->band))         &&
+		(bm_config.allband || (data->band == bandinx)) &&
+		(bm_config.allmode || (data->mode == trxmode)) &&
+		(bm_config.showdupes || !dupe) &&
+		(! bm_config.onlymults || multi)) {
+
+	    data -> dupe = dupe;
```

Comment:
Iirc idea was that the dupe state is only needed for display, grab and so on. 
But yes, setting it for all would enable us to skip the check next time we reevaluate the filter.

Reply:
I think the next check can't be avoided as the log state could have changed in the meantime. Actually, this is also true for multi info. But that one is not stored. (we have no flags to indicate that log/spots were updated and are potentially out of sync; those would help to avoid unnecessary evaluations) 

I mean more that if we don't update the dupe state of non-visible spots then on tlf_bandmap they'll have an invalid state.

Reply:
The question is: can a dupe state later be gone? Than the actual code is already bugyy. Otherwise we need to only check for dupe stations which were NOT dupe before.

Just reading your second section: Yes it needs to be done for the external tools. Will add a fix for it - not sure if I manage to do it today.

Reply:
Dupe state can be gone in a minitest.

Let's do it a separate PR. At least we have went through the potential issues a bit.